base/math.jl

```diff
@@ -41,9 +41,11 @@ end
 
 """
     clamp(x, lo, hi)
+    clamp(T, x)
```

Comment:
I think that this would be more clear with two separate docstrings instead of trying to combine them.
Something like: 
````julia
"""
    clamp(T, x)

Clamp `x` between `typemin(T)` and `typemax(T)` and convert the result to type `T`.

```jldoctest
julia> clamp(Int8, 200)
127

julia> clamp(Int8, -200)
-128
```
"""
````